### ISSUE TITLE
Make the selection drop-downs not editable 

### ISSUE THREAD

**Issue by f-odhiambo** [google/android-fhir#2228]
Current behavior: The user is able to click and edit the content of the select dropdowns in the app and alter its content 
<img src="https://github.com/google/android-fhir/assets/4540684/85945f96-d68e-4771-993b-32e88769d128" width="200" height="400" />
Need: Make the selection drop-downs not editable

**Comment by kanahia1** [comment]
Hi @f-odhiambo, I'm new to open source and want to contribute. Can you please assign me this issue.

**Comment by f-odhiambo** [comment]
@kanahia1 whats the status of this ticket ?

**Comment by qaziabubakar-vd** [comment]
@f-odhiambo I'll be picking this ticket as discussed. After reviewing the current PR, I think the cases related to the search functionality in drop-down item are yet to be worked on.

**Comment by f-odhiambo** [comment]
@qaziabubakar-vd How divergent is your PR https://github.com/google/android-fhir/pull/2393  from the previous https://github.com/google/android-fhir/pull/2256 ?

**Comment by qaziabubakar-vd** [comment]
In the [previously shared PR](https://github.com/google/android-fhir/pull/2256/files), the text input was not editable right from the start and there was no way for user to search the existing option. 
 
 In the [new draft PR](https://github.com/google/android-fhir/pull/2393) that I created, the user is able to search for the dropdown option but once an option is selected the text input is disabled. At this point the user can still select an option by expanding the dropdown but we don't have the ability to search via the text input field anymore, so the only caveat is that we need to clear the selected option using maybe a cross icon (or another approach) so that the user is able to search the dropdown options again. 
 
 cc: @f-odhiambo 

https://github.com/google/android-fhir/assets/72507786/c045beaf-4ce8-4d2c-a9f7-66a4c48e45ca

**Comment by jingtang10** [comment]
@shelaghm @rowo

i like this. thanks @qaziabubakar-vd. Can you implement the x icon in the same pr?

**Comment by qaziabubakar-vd** [comment]
> @shelaghm @rowo
> 
> i like this. thanks @qaziabubakar-vd. Can you implement the x icon in the same pr?

Yes, I can update it with the cross icon in the same PR, thanks for the feedback @jingtang10

**Comment by jingtang10** [comment]
thanks for the video - it looks beautiful!

**Comment by dubdabasoduba** [comment]
@jingtang10 @aditya-07 Please reassign this to @FikriMilano
